lib/bolt/inventory/group.rb

```diff
@@ -315,6 +316,13 @@ def resolve_data_keys(data, target = nil)
           'features' => @plugins.resolve_references(data.fetch('features', [])),
           'plugin_hooks' => @plugins.resolve_references(data.fetch('plugin_hooks', {}))
         }
+
+        # Delete keys that resolved to nil from the transport config
+        Bolt::Config::Options::TRANSPORT_CONFIG.each_key do |transport|
+          next unless result['config'].key?(transport)
+          result['config'][transport].compact!
```

Comment:
does `compact` work on nested nils?

Reply:
Do we need to account for nested nils? If so, an example would help.

Reply:
I think a common one would be:

```
config:
  ssh:
    private-key:
      key-data: 
        _plugin: foo
```
```
(base) cas@cas-ThinkPad-T460p:~/working_dir/bolt$ irb
irb(main):001:0> nested = {'private-key' => { 'key-data' => nil}}
=> {"private-key"=>{"key-data"=>nil}}
irb(main):002:0> nested.compact
=> {"private-key"=>{"key-data"=>nil}}
```

Reply:
Ok, that makes sense. Updated.